boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -206,6 +206,12 @@ static
 ULONG
 PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSize)
 {
+/* Int 15h AX=E820h Entry minimal size. */
+#define ENTRY_MIN_SIZE (size_t) 20
+
+    /* Extra (output) safety. */
+    C_ASSERT(sizeof(BIOS_MEMORY_MAP) >= ENTRY_MIN_SIZE);
```

Comment:
It would still seem more logical to me to use a FIELD_OFFSET/RTL_SIZEOF_THROUGH_FIELD here. The C_ASSERT to verify the structure is correct wrt. the API belongs next to the structure, not in the code.

Reply:
My point is: the BIOS_MEMORY_MAP structure is an abstraction for the API provided by the BIOS. The code should be able to use it and trust that it is correct without needing to validate information that's already present in the struct (in this case, the 20).
An assertion about the structure itself being correct should logically go where the structure is defined.

Hence I would use `RTL_SIZEOF_THROUGH_FIELD(BIOS_MEMORY_MAP, Type)` instead of '20' in this code, then `C_ASSERT(FIELD_OFFSET(BIOS_MEMORY_MAP, Reserved) == 20)` underneath the struct's typedef.

Reply:
Agreed. Done.